pennylane_lightning/lightning_gpu/_state_vector.py

```diff
@@ -35,13 +68,303 @@ class LightningGPUStateVector(LightningBaseStateVector):
         dtype: Datatypes for state-vector representation. Must be one of
             ``np.complex64`` or ``np.complex128``. Default is ``np.complex128``
         device_name(string): state vector device name. Options: ["lightning.gpu"]
+        mpi_handler(LightningGPU_MPIHandler): MPI handler for PennyLane Lightning GPU device.
+            Provides functionality to run on multiple devices.
+        sync (bool): immediately sync with host-sv after applying operation.
     """
 
-    def __init__(self, num_wires, dtype=np.complex128, device_name="lightning.gpu"):
-
-        if device_name != "lightning.gpu":
-            raise DeviceError(f'The device name "{device_name}" is not a valid option.')
+    def __init__(
+        self,
+        num_wires,
+        dtype=np.complex128,
+        device_name="lightning.gpu",
+        mpi_handler=None,
+        sync=True,
+    ):
 
         super().__init__(num_wires, dtype)
 
         self._device_name = device_name
+
+        if mpi_handler is None:
+            mpi_handler = LightningGPU_MPIHandler(False, 0, None, num_wires, dtype)
+
+        self._num_global_wires = mpi_handler.num_global_wires
+        self._num_local_wires = mpi_handler.num_local_wires
+
+        self._mpi_handler = mpi_handler
+        self._sync = sync
+
+        # Initialize the state vector
+        if self._mpi_handler.use_mpi:  # using MPI
+            self._qubit_state = self._state_dtype()(
+                self._mpi_handler.mpi_manager,
+                self._mpi_handler.devtag,
+                self._mpi_handler.mpi_buf_size,
+                self._mpi_handler.num_global_wires,
+                self._mpi_handler.num_local_wires,
+            )
+        else:  # without MPI
+            self._qubit_state = self._state_dtype()(self.num_wires)
+
+        self._create_basis_state(0)
+
+    def _state_dtype(self):
+        """Binding to Lightning Managed state vector C++ class.
+
+        Returns: the state vector class
+        """
+        if self._mpi_handler.use_mpi:
+            return StateVectorMPIC128 if self.dtype == np.complex128 else StateVectorMPIC64
+        else:
+            return StateVectorC128 if self.dtype == np.complex128 else StateVectorC64
+
+    def reset_state(self):
+        """Reset the device's state"""
+        # init the state vector to |00..0>
+        self._qubit_state.resetGPU(False)  # Sync reset
```

Comment:
This is not an abstract method?! 

We need to update the name of function calls across Lightning simulators. For example, `resetGPU` needs to be replaced with `resetStateVector` + update the signature of `resetStateVector` to accept 'optional' reset sync.

Reply:
Done!